File: clients/macos/FastSortClientMac/Sources/FastSortClientMac/Services/DashboardService.swift
import Foundation

struct DashboardService {
    let apiClient: APIClient

    func getCurrentUserStats() async throws -> UserStatsResponse {
        try await apiClient.request("/app/fsUserStats/getCurrentUserStats")
    }

    func getUserTagTrend(startTime: String, endTime: String) async throws -> TrendResponse {
        let body = TrendRequest(startTime: startTime, endTime: endTime)
        return try await apiClient.request("/app/fsUserStats/getUserTagTrend", body: body)
    }

    func queryRoomsByUserId(_ userId: String) async throws -> [RoomListItem] {
        try await apiClient.request("/app/fsUserRoom/queryRoomsByUserId/\(userId)")
    }

    func getAllSortBatchList(liveType: String) async throws -> SortBatchAggregate {
        let body = PageRequest(pageIndex: 1, pageSize: 1, liveType: liveType)
        return try await apiClient.request("/app/fsSortBatch/getAllFsSortBatchList", body: body)
    }

    func getBlackPage(pageIndex: Int, pageSize: Int, userId: String?) async throws -> PageResponse<BlacklistItem> {
        let body = BlacklistPageRequest(pageIndex: pageIndex, pageSize: pageSize, userId: userId)
        return try await apiClient.request("/app/fsBlack/getFsBlackPage", body: body)
    }

    func completeSortBatch(id: String, refreshIndexNumber: Int = 1) async throws {
        let body = CompleteSortBatchRequest(id: id, isRefreshIndexNumber: refreshIndexNumber)
        try await apiClient.requestVoid("/app/fsSortBatch/completeSortBatch", body: body)
    }
}

struct PageRequest: Encodable {
    let pageIndex: Int
    let pageSize: Int
    let liveType: String?
}

struct BlacklistPageRequest: Encodable {
    let pageIndex: Int
    let pageSize: Int
    let userId: String?
}

File: clients/windows/FastSort.Client.Windows/Core/Api/DashboardService.cs
using FastSort.Client.Windows.Core.Api.Dto;

namespace FastSort.Client.Windows.Core.Api;

public sealed class DashboardService
{
    private readonly ApiClient _apiClient;

    public DashboardService(ApiClient apiClient)
    {
        _apiClient = apiClient;
    }

    public Task<UserStatsResponse> GetCurrentUserStatsAsync(CancellationToken cancellationToken = default)
    {
        return _apiClient.PostAsync<UserStatsResponse>("/app/fsUserStats/getCurrentUserStats", null, cancellationToken);
    }

    public Task<TrendResponse> GetUserTagTrendAsync(string startTime, string endTime, CancellationToken cancellationToken = default)
    {
        var body = new TrendRequest(startTime, endTime);
        return _apiClient.PostAsync<TrendResponse>("/app/fsUserStats/getUserTagTrend", body, cancellationToken);
    }

    public Task<List<RoomListItem>> QueryRoomsByUserIdAsync(string userId, CancellationToken cancellationToken = default)
    {
        return _apiClient.PostAsync<List<RoomListItem>>($"/app/fsUserRoom/queryRoomsByUserId/{userId}", null, cancellationToken);
    }

    public Task<SortBatchAggregate> GetAllSortBatchListAsync(string liveType, CancellationToken cancellationToken = default)
    {
        var body = new PageRequest(1, 1, liveType);
        return _apiClient.PostAsync<SortBatchAggregate>("/app/fsSortBatch/getAllFsSortBatchList", body, cancellationToken);
    }

    public Task<PageResponse<BlacklistItem>> GetBlackPageAsync(
        int pageIndex,
        int pageSize,
        string? userId,
        CancellationToken cancellationToken = default)
    {
        var body = new BlacklistPageRequest(pageIndex, pageSize, userId);
        return _apiClient.PostAsync<PageResponse<BlacklistItem>>("/app/fsBlack/getFsBlackPage", body, cancellationToken);
    }
}
